RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -393,10 +393,19 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet * jet, f
 			    //To handle the electron case
 			    if(lPF!=nullptr) {
 			      pfTrk=(lPF->trackRef().get()==nullptr)?lPF->gsfTrackRef().get():lPF->trackRef().get();
+			      internalId_.d0_ = std::abs(pfTrk->dxy(vtx->position()));
+			      internalId_.dZ_ = std::abs(pfTrk->dz(vtx->position()));
+			    }
+			    if(lPack!=nullptr) {
```

Comment:
you already have an isPacked bool variable that you can use without needing to check the existence of the pointer every time